### PR TITLE
use navigator.mediaDevices.getUserMedia

### PR DESCRIPTION
**Pull request by fippo**
instead of getUserMedia variable from adapter.js -- a result of https://github.com/webrtc/adapter/pull/75

### CODE REVIEW

**Comment by fippo** [comment]
hrm. Actually after a quick chat with @jan-ivar I think we should go straight for navigator.mediaDevices.getUserMedia -- unless anyone objects (@samdutton?)

**Comment by jan-ivar** [comment]
> I think we should go straight for navigator.mediaDevices.getUserMedia

+1

**Comment by KaptenJansson** [comment]
>I think we should go straight for navigator.mediaDevices.getUserMedia

As per my comment in #549, please update this PR to use avigator.mediaDevices.getUserMedia instead.

Any objections @samdutton ?

**Comment by samdutton** [comment]
No objections. Agree we should use this. Will also arrange for Web Fundamentals Update explaining the change.

**Comment by fippo** [comment]
all done. Somewhat tempted to also inline the gotStream callbacks... thoughts?

**Comment by jan-ivar** [comment]
Promises facilitate continuous forward code flow, so inlining `gotStream` fits my palate, but ymmv.

**Comment by samdutton** [comment]
> Promises facilitate continuous forward code flow

Agree, but `gotStream()` is 10 lines or more in most of these, so I think it's clearer to keep it split out.

I think the split-out style is quite readable: `doSomething().then(handleResult).catch(handleError);`. 

Could be persuaded though — feel free to inline if you'd prefer :).

**Comment by fippo** [comment]
Let's merge this and fight over in-lining later (over a beer) (-:

File: src/content/peerconnection/audio/js/main.js [hunk 1]
```diff
@@ -82,14 +82,13 @@ function call() {
...
-      audio: true,
-      video: false
-    },
-    gotStream, function(e) {
-      alert('getUserMedia() error: ' + e.name);
-    });
+  navigator.mediaDevices.getUserMedia({
+    audio: true,
+    video: false
+  }).then(gotStream)
```
Comment (KaptenJansson) [inline src/content/peerconnection/audio/js/main.js]: Cannot we have `.then` on a new line like in https://github.com/webrtc/samples/pull/548/files#diff-e28becd7475fb7e59b9088adaa6eebacR60 ?

**Comment by samdutton** [comment]
LGTM with @KaptenJansson's suggestion.

File: src/content/peerconnection/audio/js/main.js (hunk 1 above)
Comment (fippo) [inline src/content/peerconnection/audio/js/main.js]: this is in several files. I have no opinion so will update

**Comment by KaptenJansson** [comment]
LGTM

**Pull request merged**